document adding custom payment information (#381)
front-commerce/developers.front-commerce.com: source/docs/advanced/payments/custom-payment-information.md

---
id: custom-payment-information
title: Custom Payment Information
---

> Implemented payment methods don't need any additional information. The user only need to select them and he will be prompted to enter all needed information. When implementing a payment method however you may require more information (Like you may want to let users enter a comment related to the payment method/checkout).

Reviewer: > The user only need to select them and he

This is not super clear, who/what is `them` referring to? 

I rephrased the ending a bit, although I think it could be improved, first need context around my question above.
```suggestion
Implemented payment methods don't need any additional information. The user only needs to select them and he will be prompted to enter all the required information. However, you may require more information, for example, if you want to let users enter a comment related to the payment method/checkout.
```

        value={comments}
        type="text"
      />
      <SubmitPayment
        gscAccepted={gscAccepted}
        onSubmit={() => {
          onAuthorize(gscAccepted ? { comments } : null);
        }}
        error={this.props.error}
      />
    </div>
  );
};

export default CustomComponent;
```

Things to understand here:

- The `SubmitPayment` component is a common component across all payment methods to ensure consistency in how methods are managed.
- `method` the current selected method.
- `value` contains initial additional data (usually it is always null).
- `onAuthorize` should be called with the additional data on `<SubmitPayment>`'s on submit method.
- `gscAccepted` whether or not the user have accepted the general sales conditions (checkbox above the "Place my order" button). Just forward this to the `<SubmitPayment>` component.
- `error` an error object if there is an error. Just forward this to the `<SubmitPayment>` component.

In this example we're only setting a comments additional text input, but you can do fancier things like fetching a list for the user to pick from GraphQL and displaying them here.

## Updating an existing method Additional Data

> Sometimes you may need to add an extra field to additional data of an existing payment method. To do so you need to first find if this payment method already has an `AdditionalDataComponent` registered. In case it does not have an `AdditionalDataComponent` registered what was in [Display a custom input for a payment method](#Display-a-custom-input-for-a-payment-method) will work fine. On the other hand if it has `AdditionalDataComponent` defined you can override the `AdditionalDataComponent` for that method and add your custom modifications there.

Reviewer: I'm not sure how checkboxes format in our docs, so just check the formatting first locally

```suggestion
Sometimes you may need to add an extra field to the additional data of an existing payment method. You first need to check if this payment method already has an `AdditionalDataComponent` registered.
- [x] If `AdditionalDataComponent` has been registered, you should then override the `AdditionalDataComponent` and add your custom modifications.
- [ ] If no `AdditionalDataComponent` is registered, you can then follow the [Display a custom input for a payment method](#Display-a-custom-input-for-a-payment-method) docs.
```
